src/server/game/Entities/Player/Player.cpp

```diff
@@ -14960,15 +14960,69 @@ void Player::RewardQuest(Quest const* quest, LootItemType rewardType, uint32 rew
         UpdatePvPState();
     }
 
-    SendQuestUpdate(quest_id);
-
     SendQuestGiverStatusMultiple();
 
+    SendQuestUpdate(quest_id, false);
+
     //lets remove flag for delayed teleports
     SetCanDelayTeleport(false);
 
+    bool canHaveNextQuest = !quest->HasFlag(QUEST_FLAGS_AUTOCOMPLETE) ? !questGiver->IsPlayer() : true;
```

Comment:
whats up with this random condition that wasnt there before?

Reply:
This condition is in QuestHandlet.cpp in a different kind of form.. we have this condition because there is a assert in CanHaveNextQuest

Reply:
Moving that code from `WorldSession::HandleQuestgiverChooseRewardOpcode` to `Player::RewardQuest` possibly changes the guid that is passed to `SendQuestGiverQuestDetails`, did you check what is the value of `packet.QuestGiverGUID` for quests with QUEST_FLAGS_AUTOCOMPLETE?